src/profilers.cpp

```diff
@@ -262,17 +269,21 @@ namespace aperf {
             nanosleep(&ts, NULL);
         }
 
+        command_status = waitpid(pid, &status, WNOHANG); //check if command to be profiled is still executing
+       
+
       }
       //return code for metric exec
-      std::exit(code_metric_exec);
-    }
+      return code_metric_exec;
+    });
 
+    int result_metric_profiler = metric_profiler.get();
 
     this->process = std::async([=]() {
       int status_metric_reader;
-      int result = waitpid(forked_metric_profiler, &status_metric_reader, 0);
+      //int result = waitpid(forked_metric_profiler, &status_metric_reader, 0);
 
-      if (result != forked_metric_profiler) {
+      if (result_metric_profiler != 0) {
```

Comment:
When you use ```metric_profiler.get()```, the result will always be your exit code (unless an exception is thrown, but this is handled elsewhere in the code), so you can remove the ```if (result_metric_profiler != 0)``` block and assign the return value of ```metric_profiler.get()``` directly to ```code```.

Reply:
Done :+1: